Expose/serve local files via the ZMQWebSocketBridge

VladimirIvan opened the pull request:
- Adds `args` argument to the initializer of `ViewerWindow`. These `args` will be passed onto the `ZMQWebSocketBridge`. `Visualizer` passes its command line arguments onto the `ViewerWindow`.
 - Adds keyboard interrupt handling for the `Visualizer`.

Reverted the below:
> - Upstreams the viewer. Now supporting loading meshes with textures.
> - Adds file handler to the `ZMQWebSocketBridge`
>     - Only known file types are served: stl, obj, dae, png, bmp, jpeg, jpg, mtl
>     - Adds `--file-root` `-f`argument. This path is the relative path for searching for files. It can be set to `/` (`-f=/`) to allow access to the whole file system (useful when exposing ROS packages).
>     - Exposes a file serving URL `http://[your_url]/files/`. Http request to this URL will strip this string and prepend the `file-root` to look for files on the server: `http://[your_url]/files/my_file.obj` will return file from `[file-root]/my_file.obj`.
>     - Adds ZMQ handler that return the file URL.

rdeits commented:
Thanks for the PR! Can you tell me more about what the purpose of the file handler is for your case? My initial reaction is that I'm uncomfortable creating such a significant security vulnerability with this tool. As written, it looks like the server will actually allow access to any file on disk to any user on the network, since a user could simply ask for a file with as many `../` in the path as necessary. Acting as a server for files on the server's machine isn't really something I intended for the meshcat server to be doing, although I'm willing to reevaluate that decision if there's a compelling reason and it can be done safely.

VladimirIvan commented:
I use the feature to expose files in ROS packages across the system. A lost of robot descriptions mix and match robot parts using XACROs so the meshes to live in various packages. These are usually in `/opt/ros/...` and in various catkin workspaces in user's home directory.

I chose to simply expose the files in favour of a more complicated and more secure solution. The file serving feature is also off by default and only mesh and texture file types are supported. 

I wanted to keep the server as independent from the the client as possible, e.g. not add too much content specific filtering. I definitely wouldn't want to add any ROS dependency.

I'm happy to add a generic white/black list feature to improve security.

VladimirIvan commented:
@rdeits I've reverted the file serving feature and moved it to a [separate repo](https://github.com/VladimirIvan/meshcat_ros_fileserver). The remaining changes are now very minimal.

rdeits commented:
Hey @VladimirIvan -- I'm sorry for the delay on this. I think the split looks like a good idea, and these changes look good to me. Thank you!
